vector/src/test/java/org/apache/arrow/vector/complex/impl/TestPromotableWriter.java

```diff
@@ -776,4 +780,95 @@ public void testPromoteToUnionFromDecimal() throws Exception {
       assertEquals(1, intHolder.value);
     }
   }
+
+  @Test
+  public void testExtensionType() throws Exception {
+    try (final NonNullableStructVector container =
+        NonNullableStructVector.empty(EMPTY_SCHEMA_PATH, allocator);
+        final UuidVector v =
+            container.addOrGet("uuid", FieldType.nullable(new UuidType()), UuidVector.class);
+        final PromotableWriter writer = new PromotableWriter(v, container)) {
+      UUID u1 = UUID.randomUUID();
+      UUID u2 = UUID.randomUUID();
+      container.allocateNew();
+      container.setValueCount(1);
+      writer.addExtensionTypeFactory(new UuidWriterFactory());
+
+      writer.setPosition(0);
+      writer.writeExtensionType(u1);
+      writer.setPosition(1);
+      writer.writeExtensionType(u2);
+
+      container.setValueCount(2);
+
+      UuidVector uuidVector = (UuidVector) container.getChild("uuid");
+      assertEquals(u1, uuidVector.getObject(0));
+      assertEquals(u2, uuidVector.getObject(1));
+    }
+  }
+
+  public class UuidWriterFactory implements ExtensionTypeWriterFactory {
+
+    @Override
+    public AbstractFieldWriter getWriterImpl(ExtensionTypeVector extensionTypeVector) {
+      if (extensionTypeVector instanceof UuidVector) {
+        return new UuidWriterImpl((UuidVector) extensionTypeVector);
+      }
+      return null;
+    }
+  }
+
+  public class UuidWriterImpl extends AbstractFieldWriter {
+    private final UuidVector vector;
+
+    public UuidWriterImpl(UuidVector vector) {
+      this.vector = vector;
+    }
+
+    @Override
```

Comment:
Is there a way we can provide default implementations for these functions to reduce the boilerplate?